resources/META-INF/federation specification schema.graphql

```diff
@@ -0,0 +1,25 @@
+# https://www.apollographql.com/docs/federation/federation-spec/#federation-schema-specification
+scalar _Any
+scalar _FieldSet
+
+# a union of all types that use the @key directive
+# noinspection GraphQLEmptyType
+union _Entity
```

Comment:
And again, graphql-java strikes back, it is completely unsuitable for use in the IDE 😞. That line breaks query validation completely because we don't do it if we have any schema errors. You suppressed this error on the file level, so it's not shown, but it still is produced by graphql-java and you could see it in the GraphQL tool window. I'll try to make a workaround, maybe I'll completely remove this validation rule.

That's why I'm not so optimistic about maintaining more and more specific frameworks in the current plugin state. There is a lot to be done to prepare a proper infrastructure for that 😞.

Reply:
Oh I see. If it's problematic, we could use the other solution of adding these directives in the JAR of the federation java library. I hadn't anticipated this being an issue, tbh. Let me know if you would prefer to remove  it altogether.

Reply:
That's not an issue with your PR, I think it's better to fix a plugin. A problem is that graphql-java "fails fast", which is an expected way to handle such errors on the server-side, but in the IDE we mostly work with invalid source code ¯\_(ツ)_/¯

I think I will simply suppress that kind of error if GraphQL Federation is enabled and ensure that a schema won't throw any unexpected NPE in random places.

Another problem I foresee is related to `directive @extends repeatable on OBJECT | INTERFACE`. AFAIU, it's expected to be used in multiple repeated object declarations like:
```
# first
type Query @extends { ... }

# one more declaration
type Query @extends { ... }
```
Initially, graphql-java silently failed in that scenario, and I spent a lot of time trying to make it work. Now we implicitly merge declarations with the same name using a straightforward "the first type/field wins" approach. But I haven't heard any feedback about how it works in real-world scenarios with Federation and similar frameworks. They allow you to declare multiple types with the same name, which will be processed somehow later, but the plugin doesn't know about that and treats those cases as a spec violation. 

Have you tried using Federation with a plugin version that includes your PR? Do you have any issues with it? Can you advise an open-source production-ready project with Apollo Federation to test it? I mean not an example starter project with 3 types, but something more complicated.

Reply:
I see. I was trying to solve a basic problem with the directives not being recognized to begin with. So the PR only addressed that by adding the definition of those directives so it doesn't complain. That's as far as I went with the scope of this change and tested with a federated schema the same way. 

As far as the use case you mentioned above, what we generally end up doing is this:
```
# first
type Query @extends { ... }

# one more declaration in the same service
extends type Query @extends { ... }
```
And that works fine. 

I don't have an open source production use case that I know of unfortunately to test this with. We do have a bunch of internal projects that I tried this with and seems ok as long as the `extends type` is used if redefining the same type in the schema.

To make it work for a large federated schema, what would be needed ultimately is to dedup the types for a composed schema. This would require some strategy around which type wins, as you mentioned, i.e., "the first type/field wins". This is not solved yet by the PR I opened, but was going to be the next issue to look into. More often than not though, a service would generally define their subgraph as the schema, so the type redefinitions would not appear in that case.

Reply:
@srinivasankavitha thanks for the details! One step I'm going to implement is to rework the resolve completely in the next few months. Now it uses a schema, so when it's broken, we have a lot of red code, but it shouldn't. The ultimate goal is to use a schema from graphql-java only for validation. Unfortunately, without that being done it doesn't make sense to work on improving the schema building process.

Reply:
Thanks, yes that makes sense to me!